src/main/kotlin/snyk/code/annotator/SnykCodeAnnotator.kt

```diff
@@ -57,17 +56,32 @@
~        }
     }
 
     fun textRange(psiFile: PsiFile, snykCodeRange: MyTextRange): TextRange {
-        val document = psiFile.viewProvider.document ?: return TextRange.EMPTY_RANGE
-        // ensure we stay within doc boundaries
-        // Snyk Code is 1-based, Document is 0-based indexes
-        val startRow = max(0, snykCodeRange.startRow - 1)
-        val endRow = max(startRow, min(document.lineCount - 1, snykCodeRange.endRow - 1))
-        val startCol = max(0, snykCodeRange.startCol)
-        val endCol = max(startCol, min(document.getLineEndOffset(endRow), snykCodeRange.endCol))
+        try {
+            val document =
+                psiFile.viewProvider.document ?: throw IllegalArgumentException("No document found for $psiFile")
+            val startRow = snykCodeRange.startRow - 1
+            val endRow = snykCodeRange.endRow - 1
+            val startCol = snykCodeRange.startCol
+            val endCol = snykCodeRange.endCol
 
-        val lineOffSet = min(document.getLineStartOffset(startRow) + startCol, document.textLength - 1)
-        val lineOffSetEnd = max(lineOffSet, min(document.getLineStartOffset(endRow) + endCol, document.textLength - 1))
-        return TextRange.create(lineOffSet, lineOffSetEnd)
+            if (startRow < 0 || startRow > document.lineCount - 1)
+                throw IllegalArgumentException("Invalid range $snykCodeRange")
+            if (endRow < 0 || endRow > document.lineCount - 1 || endRow < startRow)
+                throw IllegalArgumentException("Invalid range $snykCodeRange")
+
+            val lineOffSet = min(document.getLineStartOffset(startRow) + startCol, document.textLength - 1)
```

Comment:
let's remove min() in lines above as we have check `lineOffSet > document.textLength - 1` and `lineOffSetEnd > document.textLength - 1` later on
otherwise those checks are useless

Reply:
No, if the startCol / endCol is arbitrary high, this could still be out of bounds.

Reply:
correct, and if we do that `min()` here we'll never reach `lineOffSet > document.textLength - 1` inside `if` below and will show `lineOffSetEnd=document.textLength - 1` instead of properly proceed `throw IllegalArgumentException("Invalid range $snykCodeRange")`

Reply:
Yeah, it's true, we should throw an exception instead of using the minimum. I'll change that.

Reply:
Also, I don't know which version/branch I viewed. It makes complete sense and my comment does not make sense.